galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.manager/src/main/java/dev/galasa/cicsts/spi/BaseCicsImpl.java

```diff
@@ -93,6 +95,14 @@ public ICeda ceda() throws CicstsManagerException {
         return this.ceda;
     }
 
+    @Override
+    public ITsqHandler tsq() throws CicstsManagerException {
+        if (this.tsq == null) {
+            this.tsq = this.cicstsManager.getTsqProvider().getTsq(this, this.cicstsManager);

```

Comment:
Isn't this creating the TSQ rather than just getting it ?

Reply:
Hi @techcobweb , This is not creating a TSQ, it is creating a ITsqHandler instance for the CICS region which has methods that can be used to create a TSQ (read, write, delete, make recoverable).